contracts/validator/ValidatorSet.sol

```diff
@@ -50,4 +50,80 @@ contract ValidatorSet {
      *   - less than MAX_FUTURE_END_HEIGHT: for logged out validators.
      */
     mapping(address => uint256) public validatorEndHeight;
+
+
+    /* Internal functions  */
+
+    /**
+     * @notice It is for inserting validators into the validator set.
+     *
+     * @dev Function requires :
+     *          - Validator address must not be 0.
+     *          - Validator address is already used.
+     *
+     * @param _validator Validator address.
+     * @param _beginHeight Begin height for the validator.
+     */
+    function insertValidator(
+        address _validator,
+        uint256 _beginHeight
+    )
+        internal
+    {
+        require(
+            _validator != address(0),
+            "Validator must not be null address."
+        );
+        require(
+            validatorBeginHeight[_validator] == 0,
+            "Validator address is already used."
+        );
+
+        // First validator is being added.
```

Comment:
Should we do this during a construction/setup?

Reply:
If we do it during construction/setup, we would need to have validator address. Also, not sure how the flow will be from Core/Protocore.
@deepesh-kn and  @jayeshbairagi, what do you guys think ?

Reply:
As discussed with @deepesh-kn , We will have **setup** method in ValidatorSet.
In **setup** method, we would initialize **validator** mapping as below :
```
validator[SENTINEL_VALIDATORS] = SENTINEL_VALIDATORS;
``` 
We can now remove condition of first validator being added.

Reply:
@gulshanvasnani have we removed the condition of first validator being added?